domain/library/src/main/java/com/into/websoso/domain/library/model/ReadStatus.kt

```
@@ -0,0 +1,14 @@
+package com.into.websoso.domain.library.model
+
+enum class ReadStatus {
+    WATCHING,
+    WATCHED,
+    QUIT,
+    ;
+
+    companion object {
+        fun valueOf(name: String): ReadStatus =
+            entries.find { readStatus -> readStatus.name == name }
+                ?: throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)